pkg/controller/ingress/ingress_controller_test.go

```diff
@@ -54,6 +60,88 @@ var (
 	}
 )
 
+func TestRouteMigration(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+
+	test := struct {
+		name            string
+		exRoute         *routev1.Route
+		noLabelRoute    *routev1.Route
+		otherLabelRoute *routev1.Route
+		wantName        string
+	}{
+		name: "Clean up old route and new route is generated",
+		exRoute: &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "test-0",
+				Labels: map[string]string{networking.IngressLabelKey: name, serving.RouteLabelKey: name, serving.RouteNamespaceLabelKey: namespace},
+			},
+			Spec: routev1.RouteSpec{Host: domainName},
+		},
+		noLabelRoute: &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "no-remove-1",
+				Labels: map[string]string{networking.IngressLabelKey: name, serving.RouteLabelKey: name},
+			},
+			Spec: routev1.RouteSpec{Host: domainName},
+		},
+		otherLabelRoute: &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "no-remove-2",
+				Labels: map[string]string{networking.IngressLabelKey: "another", serving.RouteLabelKey: name, serving.RouteNamespaceLabelKey: namespace},
+			},
+			Spec: routev1.RouteSpec{Host: domainName},
+		},
+		wantName: routeName0,
+	}
+
+	t.Run(test.name, func(t *testing.T) {
+		// Register operator types with the runtime scheme.
+		s := scheme.Scheme
+		s.AddKnownTypes(networkingv1alpha1.SchemeGroupVersion, defaultIngress)
+		s.AddKnownTypes(routev1.SchemeGroupVersion, test.exRoute)
+		s.AddKnownTypes(routev1.SchemeGroupVersion, &routev1.RouteList{})
+
+		// Create a fake client to mock API calls.
+		cl := fake.NewFakeClient(defaultIngress, test.exRoute, test.noLabelRoute, test.otherLabelRoute)
+
+		// Create a Reconcile Ingress object with the scheme and fake client.
+		r := &ReconcileIngress{base: &common.BaseIngressReconciler{Client: cl}, client: cl, scheme: s}
+		// Mock request to simulate Reconcile() being called on an event for a
+		// watched resource .
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+		if _, err := r.Reconcile(req); err != nil {
+			t.Fatalf("reconcile: (%v)", err)
+		}
+
+		route := &routev1.Route{}
```

Comment:
Per the comment above, we should be able to replace all of this via a check against a „want“ list of routes